website/src/components/edit-word-audio/contributor.tsx

```diff
@@ -121,35 +154,70 @@ export function ContributorEditWordAudio(p: {
   )
 
   return (
-    <div>
+    <>
       {[curatedAudioContent, userAudioContent, otherUserAudioContent]
         .filter((content) => content !== false)
         .map((content, idx) => (
-          <>
+          <Fragment key={idx}>
             {idx > 0 && <hr />}
             {content}
-          </>
+          </Fragment>
         ))}
+    </>
+  )
+}
+
+function ContributeAudioSection(p: {
+  word: Dailp.FormFieldsFragment
+}): ReactElement {
+  const [currentTab, setCurrentTab] = useState<"upload" | "record">()
+  const [selectedFile, setSelectedFile] = useState<File>()
+  const [uploadAudio, uploadState, clearUploadError] = useAudioUpload(p.word.id)
+
+  function onFileChanged(event: ChangeEvent<HTMLInputElement>) {
+    if (!event.currentTarget.files) return
+
+    const [file] = event.currentTarget.files
+    if (!file) return
+    console.log("New file selected", file)
+    setSelectedFile(file)
+    setCurrentTab("upload")
+  }
+
+  async function uploadAudioAndReset(data: Blob) {
+    const success = await uploadAudio(data)
+    if (success) {
```

Comment:
Should we notify the user if this fails, or at least log an error in the console?